From 0412a9d22182ea7988dfad11b8194fc5249b5bcf Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 31 May 2024 22:54:37 +0200
Subject: [PATCH 32/42] mediatek: mt7981: define -u-boot DTSI to use upstream
 linux kernel DTSI

Define mt7981-u-boot.dtsi to use upstream linux kernel DTSI. This
contains all the required property specific to uboot.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 arch/arm/dts/mt7981-u-boot.dtsi | 71 +++++++++++++++++++++++++++++++++
 1 file changed, 71 insertions(+)
 create mode 100644 arch/arm/dts/mt7981-u-boot.dtsi

diff --git a/arch/arm/dts/mt7981-u-boot.dtsi b/arch/arm/dts/mt7981-u-boot.dtsi
new file mode 100644
index 00000000000..eded3b08b51
--- /dev/null
+++ b/arch/arm/dts/mt7981-u-boot.dtsi
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2022 MediaTek Inc.
+ * Author: Christian Marangi <ansuelsmth@gmail.com>
+ */
+
+#include <dt-bindings/reset/mt7629-reset.h>
+
+/ {
+	cpus {
+		cpu@0 {
+			mediatek,hwver = <&hwver>;
+		};
+
+		cpu@1 {
+			mediatek,hwver = <&hwver>;
+		};
+	};
+
+	gpt_clk: gpt_dummy20m {
+		compatible = "fixed-clock";
+		clock-frequency = <13000000>;
+		#clock-cells = <0>;
+		bootph-all;
+	};
+
+	soc {
+		hwver: hwver {
+			compatible = "mediatek,hwver", "syscon";
+			reg = <0 0x8000000 0 0x1000>;
+		};
+
+		timer0: timer@10008000 {
+			compatible = "mediatek,mt7986-timer";
+			reg = <0 0x10008000 0 0x1000>;
+			interrupts = <GIC_SPI 130 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&gpt_clk>;
+			clock-names = "gpt-clk";
+			bootph-all;
+		};
+	};
+
+	timer {
+		arm,cpu-registers-not-fw-configured;
+	};
+};
+
+&apmixedsys {
+	bootph-all;
+};
+
+&topckgen {
+	bootph-all;
+};
+
+&infracfg {
+	bootph-all;
+};
+
+&uart0_pins {
+	bootph-all;
+};
+
+&uart0 {
+	bootph-all;
+};
+
+&eth {
+	resets = <&ethsys ETHSYS_FE_RST>;
+	reset-names = "fe";
+};
-- 
2.43.0

